src/signal.cr

```diff
@@ -324,9 +324,11 @@ fun __crystal_sigfault_handler(sig : LibC::Int, addr : Void*)
   # Determine if the SEGV was inside or 'near' the top of the stack
   # to check for potential stack overflow. 'Near' is a small
   # amount larger than a typical stack frame, 4096 bytes here.
-  stack_top = Pointer(Void).new(Fiber.current.@stack.address - 4096)
 
-  if stack_top <= addr < Fiber.current.@stack_bottom
+  stack = Fiber.current.stack
+  stack_top = Pointer(Void).new(stack.pointer(0).address - 4096)
```

Comment:
you can perform arithmetic on pointers directly.

Reply:
just make it a named local variable, like `page_size = 4096` then `stack.pointer(0) - page_size`. Constants don't need to be constants in the language if they're only used in one function. They just need to be constant.

Reply:
Addressed via a local variable.

@RX14 , my first experience with pointer arithmetic in Crystal was with subtracting a `UInt64`, which is natural to figure out a stack top when you know the bottom and the stack grows down.  But this seems to be broken, and I shied away from it afterwards.  But I think it's just subtracting a `UInt` that's problematic and could be fixed in a separate PR.